src/queries.jl

```diff
@@ -397,14 +397,30 @@ julia> findfreeslot(reg) |> isnothing
 true
 ```
 """
-function findfreeslot(reg::Register; randomize=false, margin=0)
+function findfreeslot(reg::Register; filter=minimum::Union{Int,<:Function}, randomize=false, margin=0)
```

Comment:
let's rename `filter` to `choose` -- to me "filter" sounds like it might return more than one option